File: internal/domain/entity/club.go
package entity

import "time"

type Club struct {
	ID                int64     `json:"id"                  db:"id"`
	Name              string    `json:"name"                db:"name"`
	TelegramChatID    int64     `json:"telegram_chat_id"    db:"telegram_chat_id"`
	WelcomeEnabled    bool      `json:"welcome_enabled"     db:"welcome_enabled"`
	BirthdayEnabled   bool      `json:"birthday_enabled"    db:"birthday_enabled"`
	RaceNotifyEnabled bool      `json:"race_notify_enabled" db:"race_notify_enabled"`
	CreatedAt         time.Time `json:"created_at"          db:"created_at"`
	UpdatedAt         time.Time `json:"updated_at"          db:"updated_at"`
}

File: web/admin/src/api/club.ts
import client from './client'

export interface Club {
  id: number
  name: string
  telegram_chat_id: number
  welcome_enabled: boolean
  birthday_enabled: boolean
  race_notify_enabled: boolean
  created_at?: string
  updated_at?: string
}

export interface ClubCreate {
  name: string
  telegram_chat_id: number
  welcome_enabled?: boolean
  birthday_enabled?: boolean
  race_notify_enabled?: boolean
}

export interface ClubUpdate {
  name?: string
  telegram_chat_id?: number
  welcome_enabled?: boolean
  birthday_enabled?: boolean
  race_notify_enabled?: boolean
}

export default {
  list() {
    return client.get<Club[]>('/clubs')
  },

  create(data: ClubCreate) {
    return client.post<Club>('/clubs', data)
  },

  get(id: number) {
    return client.get<Club>(`/clubs/${id}`)
  },

  update(id: number, data: ClubUpdate) {
    return client.put<Club>(`/clubs/${id}`, data)
  },

  remove(id: number) {
    return client.delete(`/clubs/${id}`)
  },
}
